pkg/management/postgres/webserver/database.go

```diff
@@ -0,0 +1,293 @@
+			log.Debug(
+				"Error while changing database ownership",
+				"dbname", dbname,
+				"err", err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	if database.IsTemplate != nil {
+		changeIsTemplateSQL := fmt.Sprintf(
+			"ALTER DATABASE %s WITH IS_TEMPLATE %v",
+			pgx.Identifier{dbname}.Sanitize(),
+			*database.IsTemplate)
+
+		if _, err := db.ExecContext(req.Context(), changeIsTemplateSQL); err != nil {
+			log.Debug(
+				"Error while changing database IS_TEMPLATE option",
+				"dbname", dbname,
+				"err", err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	if database.AllowConnections != nil {
+		changeAllowConnectionsSQL := fmt.Sprintf(
+			"ALTER DATABASE %s WITH ALLOW_CONNECTIONS %v",
+			pgx.Identifier{dbname}.Sanitize(),
+			*database.AllowConnections)
+
+		if _, err := db.ExecContext(req.Context(), changeAllowConnectionsSQL); err != nil {
+			log.Debug(
+				"Error while changing database ALLOW_CONNECTIONS option",
+				"dbname", dbname,
+				"err", err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	if database.ConnectionLimit != nil {
+		changeConnectionsLimitSQL := fmt.Sprintf(
+			"ALTER DATABASE %s WITH CONNECTION LIMIT %v",
+			pgx.Identifier{dbname}.Sanitize(),
+			*database.ConnectionLimit)
+
+		if _, err := db.ExecContext(req.Context(), changeConnectionsLimitSQL); err != nil {
+			log.Debug(
+				"Error while changing database CONNECTION LIMIT option",
+				"dbname", dbname,
+				"err", err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	if len(database.Tablespace) > 0 {
+		changeTablespaceSQL := fmt.Sprintf(
+			"ALTER DATABASE %s SET TABLESPACE %s",
+			pgx.Identifier{dbname}.Sanitize(),
+			pgx.Identifier{database.Tablespace}.Sanitize())
+
+		if _, err := db.ExecContext(req.Context(), changeTablespaceSQL); err != nil {
+			log.Debug(
+				"Error while changing database default tablespace",
+				"dbname", dbname,
+				"err", err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func (ws *remoteWebserverEndpoints) putPgDatabase(w http.ResponseWriter, req *http.Request) {
+	dbname := req.PathValue("dbname")
+
+	var database pgDatabaseRequestFormat
+	if err := json.NewDecoder(req.Body).Decode(&database); err != nil {
+		log.Debug("Error while decoding patch database request",
+			"dbname", dbname,
+			"error", err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	db, err := ws.instance.GetSuperUserDB()
+	if err != nil {
+		log.Debug(
+			"Error while getting DB connection",
+			"dbname", dbname,
+			"err", err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	sqlCreateDatabase := fmt.Sprintf("CREATE DATABASE %s ", dbname)
```

Comment:
this being a PUT, we might get it for an existing database, so we'd also need to have ALTER statements if the DB already exists, no?
I know the PATCH is the right verb for that.
So maybe if we try to PUT a DB that already exists, we could reject it.

Reply:
PATCH is already implemented, too. We could keep the one we like best between PATCH and PUT and use just that.

Reply:
Resolving this as this PR is not using the webserver anymore.